src/patternfly/layouts/Flex/flex.scss

```diff
@@ -311,5 +332,75 @@ $pf-l-flex--variable-map: build-variable-map("pf-l-flex--spacer", $pf-l-flex--sp
       }
     }
   }
+
+
+  // .pf-m-gap-{size}{-on-breakpoint}
+  @each $breakpoint, $breakpoint-value in $pf-l-flex--breakpoint-map {
+    $breakpoint-name: if($breakpoint != "base", -on-#{$breakpoint}, "");
+
+    @include pf-apply-breakpoint($breakpoint) {
+      @each $spacer, $spacer-value in $pf-l-flex--spacer-map {
+        &.pf-m-gap#{$breakpoint-name} {
+          --pf-l-flex--ColumnGap: var(--pf-l-flex--ColumnGap-base);
+          --pf-l-flex--RowGap: var(--pf-l-flex--RowGap-base);
+
+          > * {
+            --pf-l-flex--spacer: 0;
+          }
+        }
+        &.pf-m-gap-#{$spacer}#{$breakpoint-name} {
+          --pf-l-flex--spacer-base: 0;
+          --pf-l-flex--RowGap: var(#{map-get($pf-l-flex--variable-map, $spacer-value)});
+          --pf-l-flex--ColumnGap: var(#{map-get($pf-l-flex--variable-map, $spacer-value)});
+
+          > * {
+            --pf-l-flex--spacer: 0;
+          }
+        }
```

Comment:
With all of these loops, having both `.pf-m-[row/column-]gap#{$breakpiont-name}` and `.pf-m-[row/column-]gap-#{$spacer}#{$breakpoint-name}` is causing the `.pf-m-[row/column-]gap#{$breakpiont-name}` block to be duplicated in each iteration of the spacer map loop. I _think_ you could just move that block outside/before the spacer loop.

Here are a couple of screenshots.

<img width="416" alt="Screenshot 2023-01-04 at 9 15 37 PM" src="https://user-images.githubusercontent.com/35148959/210693660-655a4334-3ee1-4224-af83-cda794be0c60.png">
<img width="462" alt="Screenshot 2023-01-04 at 9 15 16 PM" src="https://user-images.githubusercontent.com/35148959/210693675-e49f1bf2-7204-4226-b20a-5388b83514a8.png">